File: app/src/main/java/com/example/apprifa/Controlers/Reset_senha.java
package com.example.apprifa.Controlers;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;

import com.example.apprifa.Helpers.AccessFirebase;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.example.apprifa.R;

public class Reset_senha extends AppCompatActivity {

    EditText email_reset;
    Button reset_senha;

    ProgressDialog prg_reset_senha;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reset_senha);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        reset_senha = findViewById(R.id.btn_alterar_senha);
        email_reset = findViewById(R.id.ed_reset_email);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);

        prg_reset_senha = new ProgressDialog(Reset_senha.this);

        setTitle("Trocar senha");

        reset_senha.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                prg_reset_senha.setMessage("Enviando...");
                prg_reset_senha.show();

                new AccessFirebase().reset_senha(email_reset.getText().toString(),Reset_senha.this);

                prg_reset_senha.dismiss();

                Intent i_reset = new Intent(getApplicationContext(),EntrarUsuario.class);
                startActivity(i_reset);
                finish();
            }
        });


    }

    public boolean onOptionsItemSelected(MenuItem item) { //Botão adicional na ToolBar
        switch (item.getItemId()) {
            case android.R.id.home:  //ID do seu botão (gerado automaticamente pelo android, usando como está, deve funcionar
                startActivity(new Intent(getApplicationContext(), EntrarUsuario.class));  //O efeito ao ser pressionado do botão (no caso abre a activity)
                finish();  //Método para matar a activity e não deixa-lá indexada na pilhagem
                break;
            default:
                break;
        }
        return true;
    }

}

File: app/src/main/java/br/com/medeve/Activitys/ResetSenha.kt
package br.com.medeve.Activitys

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import br.com.medeve.Models.Usuario
import br.com.medeve.R
import br.com.medeve.Util.Constantes
import br.com.medeve.ViewModels.UsuarioViewModel
import kotlinx.android.synthetic.main.activity_reset_senha.*
import org.koin.androidx.viewmodel.ext.android.viewModel

class ResetSenha : AppCompatActivity(), View.OnClickListener {

    val usuarioViewModel: UsuarioViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reset_senha)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setHomeButtonEnabled(true)
        title = "Trocar senha"

        btn_alterar_senha.setOnClickListener(this)

        setObservers()
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.btn_alterar_senha -> {
                val usuario = Usuario()
                usuario.email = ed_reset_email.text.toString()
                if (usuario.email.isNullOrEmpty()) {
                    Toast.makeText(this, "Informe o seu e-mail.", Toast.LENGTH_LONG).show()
                } else {
                    usuarioViewModel.resetSenhaUsuario(usuario)
                }
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                startActivity(
                    Intent(
                        applicationContext,
                        EntrarUsuarioActView::class.java
                    )
                )
                finish()
            }
            else -> {
            }
        }
        return true
    }

    fun setObservers() {
        usuarioViewModel.getResetSenhaUsuarioMutableLiveData().observe(this, {
            when (it) {
                Constantes.ResetSenha.RESET_SENHA_SUCESSO -> {
                    Toast.makeText(
                        this,
                        "E-mail de reset de senha enviado para ${ed_reset_email.text.toString()}",
                        Toast.LENGTH_LONG
                    )
                        .show()
                }
                Constantes.ResetSenha.EMAIL_INVALIDO -> {
                    Toast.makeText(this, "E-mail Inválido", Toast.LENGTH_LONG)
                        .show()
                }
                Constantes.ResetSenha.ERRO_ENVIO_EMAIL -> {
                    Toast.makeText(this, "Erro ao enviar o e-mail.", Toast.LENGTH_LONG)
                        .show()
                }
            }
        })
    }


}

